### PR TITLE
Added AzureCliCredential to possible Key Vault authentication methods

### LINKED ISSUE #20710: [FEATURE REQ] Add AzureCliCredential to possible Key Vault authentication methods
**Is your feature request related to a problem? Please describe.**

When developing locally, I would like to re-use my `az login` when running my Spring Boot application. This way my application can access secrets without including any secrets in the configuration (requiring jasypt or similar).

I would also imagine this to be useful for console applications.

**Describe the solution you'd like**
I would like `com.azure.spring.keyvault.KeyVaultEnvironmentPostProcessorHelper#getCredentials(java.lang.String)` to include `com.azure.identity.AzureCliCredential` as an option. Probably enabled by something like `azure.keyvault.enable-cli-auth`.

**Describe alternatives you've considered**
An alternativ would be to override beans, but it seems the Azure Key Vault integration is not using things like `@ConditionalOnMissingBean`. I guess that is because it is too early in the spring boot startup process?

Another option would be to require the developer to create a service principal and add it as environment variables, but that is not as elegant as `az login`.

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Description Added
- [X] Expected solution specified

### PR DESCRIPTION
**Pull request by anderius**
Fixes #20710.

### CODE REVIEW

File: sdk/spring/azure-spring-boot-starter-keyvault-secrets/README.md [hunk 1]
```diff
@@ -72,6 +72,7 @@ azure.keyvault.refresh-interval=1800000
 azure.keyvault.secret-keys=key1,key2,key3
 azure.keyvault.authority-host=put-your-own-authority-host-here(fill with default value if empty)
 azure.keyvault.secret-service-version=specify secretServiceVersion value(fill with default value if empty)
+azure.keyvaul.cli-credentials-enabled=false
```
Comment (christophernruud) [inline sdk/spring/azure-spring-boot-starter-keyvault-secrets/README.md:75]:
```suggestion
azure.keyvault.cli-credentials-enabled=false
```

**Review by christophernruud — COMMENTED** [review]
Small typo in conf

File: sdk/spring/azure-spring-boot-starter-keyvault-secrets/README.md (hunk 1 above)
Comment (anderius) [reply sdk/spring/azure-spring-boot-starter-keyvault-secrets/README.md]: Thank you, @christophernruud!

**Comment by stliu** [comment]
How about we use _com.azure.identity.DefaultAzureCredentialBuilder_ instead? so, this can find the credential with specific order w/ explicitly enable/disable cli  

https://docs.microsoft.com/en-us/java/api/com.azure.identity.defaultazurecredential?view=azure-java-stable

**Comment by anderius** [comment]
@stliu Very good idea, I did not know about that class. We will have to check if the change is backward compatible, maybe?

**Comment by anderius** [comment]
@stliu I don't see how we can do that and be backwards compatible. The class KeyVaultEnvironmentPostProcessorHelper reads configuration properties like `client-key`, and I don't see a way to set that on DefaultAzureCredentialBuilder (which seems to only support environment variables).

Any hint?

**Comment by stliu** [comment]
@anderius what about we create a _SpringAzureCredentialBuilder_ which functions simliar as _DefaultAzureCredentialBuilder_ but read properties from sprint env? Basicaly I'm thinking we can extract _com.azure.spring.keyvault.KeyVaultEnvironmentPostProcessorHelper#getCredentials(java.lang.String)_ from the _KeyVaultEnvironmentPostProcessorHelper_

**Comment by stliu** [comment]
@saragluna @chenrujun @ZhijunZhao thoughts ?

**Comment by anderius** [comment]
I think that is a good idea. Let me know if the others have other thoughts, and I will be happy to adjust the PR.

**Comment by chenrujun** [comment]
@anderius 

> I don't see how we can do that and be backwards compatible. The class KeyVaultEnvironmentPostProcessorHelper reads configuration properties like client-key, and I don't see a way to set that on DefaultAzureCredentialBuilder (which seems to only support environment variables).

We can write a method like this:
https://github.com/Azure/azure-sdk-for-java/blob/azure-spring-boot-starter-keyvault-certificates_3.0.0-beta.6/sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java#L75

```
    private void putEnvironmentPropertyToSystemProperty(ConfigurableEnvironment environment, String key) {
        Optional.of(key)
                .map(environment::getProperty)
                .filter(StringUtils::hasText)
                .ifPresent(value -> System.getProperties().put(key, value));
    }
```

**Comment by chenrujun** [comment]
@stliu 
If we use `putEnvironmentPropertyToSystemProperty`, maybe we can use `DefaultAzureCredentialBuilder` directly instead of create `SpringAzureCredentialBuilder`.

**Comment by stliu** [comment]
I'd more prefer to have a customized _SpringAzureCredentialBuilder_ instead of some kind of hack that push environmetn to system properties

**Comment by anderius** [comment]
Agreed. If we overwrite system properties, we overwrite the standard way to overwrite configuration.

I will make an attempt with a SpringAzureCredentialBuilder.

**Comment by anderius** [comment]
I was pressed on time, so if anyone will make an attempt, feel free!

I am still not sure about how we make the change backwards compatible.

**Comment by anderius** [comment]
I will not have time to pursue this any further.

**Comment by saragluna** [comment]
This issue will be addressed in this PR https://github.com/Azure/azure-sdk-for-java/pull/22472.

**Pull request closed without merging**